packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -173,4 +183,116 @@ export class MultichainAccountWallet<
   getMultichainAccountGroups(): MultichainAccountGroup<Account>[] {
     return Array.from(this.#accounts.values()); // TODO: Prevent copy here.
   }
+
+  /**
+   * Gets next group index for this wallet.
+   *
+   * @returns The next group index of this wallet.
+   */
+  getNextGroupIndex(): number {
+    // We do not check for gaps.
+    return (
+      Math.max(
+        -1, // So it will default to 0 if no groups.
+        ...this.#accounts.keys(),
```

Comment:
Nit: We should probably rename this to `accountGroups` as the name is more descriptive. Accounts can be misunderstood to be a singular address.